jaeger-core/src/main/java/com/uber/jaeger/utils/Utils.java

```diff
@@ -50,7 +50,7 @@ public static int ipToInt(String ip) throws EmptyIpException, NotFourOctetsExcep
 
     int intIp = 0;
     for (byte octet : octets.getAddress()) {
-      intIp = (intIp << 8) | (octet);
+      intIp = (intIp << 8) | (octet & 0xFF);
```

Comment:
for my education, since `octet` is already of type `byte`, why does `& 0xFF` make a difference?